### DISCUSSION TITLE
Better UI for construction windows

### DISCUSSION THREAD

**Discussion by victorleblais** [voxeltycoon/issues#378] [UI&UX]
**Is your feature request related to a problem? Please describe.**
The UI, despite being very beautiful, is also quite basic when it comes to building.

**Describe the solution you'd like**
For all building windows (in this example: Decorations), but it could also be extended to Factories, Stations, Warehouses, etc...), we could use some additional features:
- Filter by item name
- Change to look of the list (either grid or list)
- Change the size of the images to avoid scrolling a lot
- Filter by mod (which will become exponentially required !)
- Additional options for comfort of building (see #94 and #95)

The ability to **resize** windows would be cool too.

**Describe alternatives you've considered**
My take on the UI is just that: an idea. Right now there isn't any alternative in game except scrolling way too much. Feel free to comment on what could be useful.

**Additional context**
Here's what the window could look like:
![MainWindow](https://user-images.githubusercontent.com/4365075/60738666-6c591e00-9f5f-11e9-80e2-b4387174b8ea.png)

Clicking on the filter would open this:
![Filter](https://user-images.githubusercontent.com/4365075/60738676-73802c00-9f5f-11e9-8fd0-df1b9acb96bf.png)

And clicking on options would open this:
![Options](https://user-images.githubusercontent.com/4365075/60738686-7a0ea380-9f5f-11e9-9f9d-51eb86bb6f7a.png)

I took some random icons, everything is misaligned, colors are incorrect.... But you get the idea :p

**Comment by katepolak** [comment]
#13 was closed as the exact thing it suggested was implemented, but it would still be nice to improve on it further as described here.

**Comment by int3ger** [comment]
Add a Favorites feature to this so we can create custom named lists of often used items, or a simple toggle to show/hide those 
favourited items.
